### DISCUSSION TITLE
Difficulty with updating my branch with main

### DISCUSSION THREAD

**Discussion by annaposlednik** [earthlab-education/Earth-Analytics-AY24#268] [Q&A] [answered]
Hi, I'm trying to update my branch and received this error message: 
<img width="1002" alt="Screenshot 2024-09-05 at 2 04 14 PM" src="https://github.com/user-attachments/assets/cc893b9a-65b4-4dd3-a0f5-c6325d4f1e84">

Any ideas what to do?

**Comment by akpost21** [comment]
@eculler Could you please assist with this?

**Reply by eculler** [reply]
Hi @annaposlednik, you don't need to worry about this. Because we're all editing close to each other in the same file, we often run into this issue for this activity, as well as the home towns activity for this week. Basically, git doesn't have any way to tell what order to out the rows in, so I will go in and tell it once you have two approving reviews and we're ready to merge.

**Comment by eculler** [comment]
Oh, sorry I re-read this and realized you were trying to update your branch with main.  Hm...I think this might resolve itself once we have your Pull Request merged. If you see that your pull request has been merged, but you still have conflicts, we can help you resolve them. In the meantime, if you want to you can try out resolving the conflicts yourself. First, you click on the Resolve Merge Conflicts button. That will open a text editor for you online.

git will indicate to you which parts of a file is conflicting, something like:

```
>>>>>>> your-branch-name
Some changes
=======
Some other changes
<<<<<<< main
```

When you resolve the changes, you can keep both lines, one line of your choice, or a new line that combines them. Then, you delete all the special characters (<,>,=) and confirm that you want to resolve the merge.

You can also check out the [Software Carpentries lesson on resolving merge conflicts](https://swcarpentry.github.io/git-novice/09-conflict.html) (although they are working on a local computer so it will be a little different)

Just know that you will continue to have conflicts in the future if your branch is different from main!

**Comment by hanried** [comment, marked as answer]
Hey @annaposlednik, I was also having this issue in class on Thursday and I think I was just able to fix it. 
I saw the same thing as your screenshot in my merge pull request to merge the main branch to my branch. I noticed the "Resolve Conflicts" button was greyed out so I couldn't click on it, and I think that's the "Resolve Merge Conflicts Button" Elsa is referring to.

So instead, I went to the pull request I submitted to update the README.md w/ my new username (#228). This pull request hadn't been fully approved and merged w/ the main branch yet so I figured that's where the conflicts were coming from. I scrolled down to the bottom and saw a similar "This branch has conflicts" message but I was able to click on the "Resolve Conflicts" button there and fix the conflicts as Elsa described. I think my branch is also now updated with all main branch changes.

It looks like you also have an open pull request for updating the README.md (#86). So maybe try resolving conflicts in your #86 pull request and that will let you merge the main branch to yours.

Sorry I don't have screenshots, I wasn't fully sure whether I was fixing things while doing it so I didn't think to take screenshots.

**Reply by annaposlednik** [reply]
Thank you so much!!!

**Comment by ZacharyCJDecker** [comment]
Hi all

I am having the same issue. I was following hanried's steps but when I get to my README.md the option to resolve conflicts is greyed out. 

Perhaps this needs to be approved first? It is still missing two approvals as far as I can tell.

**Reply by ZacharyCJDecker** [reply]
I believe I solved the problem. 

I could not resolve conflicts when viewing the README.md from the 'commits behind'. However, in the 'commits ahead' I could resolve the conflicts. 

I don't yet understand why that is, but it fixed the issue.

**Comment by eculler** [comment]
Nice troubleshooting here, all! Conflicts in git are very tricky, and I'm glad to see you're all on the way to becoming experts. I am going try to merge all the pull requests for the syllabus in the next few days whether or not the approving review are in, which I think will resolve some of the conflicts with the updated main branch for you. We get those conflicts because we're all editing very close together in the same file, and so they're less likely to come up later in the semester (phew!).

@byandell if you end up here...you are a special case because you have conflicts that can not be resolved on GitHub! I will look at those over the next few days and see if I can resolve using fancier git, and will let you know if I figure out the issue.
